lib/resty/timerng/thread/loop.lua

```diff
@@ -97,10 +100,11 @@ end
 ---@return integer action
 ---@return string message
 local function phase_handler_wrapper(self, phase)
+    -- unpack arguments to avoid NYI: return to lower frame
+    -- as `pcall` with varargs might causes NYI when returning from `pcall`
+    local a0, a1, a2, a3 = table_unpack(self[phase].argv, 1, self[phase].argc)
```

Comment:
Only one suggestion:
Should we name these variables from `1` to follow Lua's tradition.